huggingface_pipelines/text.py

```diff
@@ -0,0 +1,161 @@
+            embeddings = batch[f"{column}_embeddings"]
+            logger.info(f"Embeddings: {embeddings}")
+            reconstructed_texts = self.decode_embeddings(embeddings)
+            logger.info(f"Reconstructed Texts: {reconstructed_texts}")
+            batch[f"{column}_reconstructed"] = reconstructed_texts
+        return batch
+
+    def decode_embeddings(self, embeddings: List[Any]) -> List[str]:
+        try:
+            logger.info(f"Decoding {len(embeddings)} embeddings...")
+
+            # Convert embeddings to tensors if they're not already
+            embeddings = [torch.tensor(embed) if not isinstance(
+                embed, torch.Tensor) else embed for embed in embeddings]
+
+            # Process each embedding separately
+            decoded_texts = []
+            for embed in embeddings:
+                if embed.dim() == 1:
+                    embed = embed.unsqueeze(0)
+
+                # Decode the embedding
+                decoded = self.t2t_model.predict(
+                    embed, target_lang=self.config.target_lang, batch_size=self.config.batch_size)
+
+                decoded_texts.append(decoded)
+
+            logger.info("Texts decoded successfully.")
+            return decoded_texts
+        except Exception as e:
+            logger.error(f"Error decoding texts: {str(e)}")
+            logger.error(f"Error type: {type(e).__name__}")
+            raise
+
+
+@dataclass
+class HFTextToEmbeddingPipeline(Pipeline):
+    config: TextToEmbeddingPipelineConfig
+
+    def __post_init__(self):
+        logger.info("Initializing text to embedding model...")
+        self.t2vec_model = TextToEmbeddingModelPipeline(
+            encoder=self.config.encoder_model,
+            tokenizer=self.config.encoder_model,
+            device=self.config.device
+        )
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, Any]:
+        for column in self.config.columns:
+            if column in batch:
+                sentence_embeddings = []
+                for item in batch[column]:
+                    if isinstance(item, list):
+                        # If it's a list of sentences, encode each sentence
+                        embeddings = self.encode_texts(item)
+                    else:
+                        # If it's a single string, treat it as one sentence
+                        embeddings = self.encode_texts([item])
+                    sentence_embeddings.append(embeddings)
+                batch[f"{column}_embeddings"] = sentence_embeddings
+            else:
+                logger.warning(f"Column {column} not found in batch.")
+        return batch
+
+    def encode_texts(self, texts: Union[str, List[str]]) -> List[np.ndarray]:
+        try:
+            if isinstance(texts, str):
+                texts = [texts]
+            elif not isinstance(texts, list):
+                raise ValueError(
+                    f"Expected string or list of strings, got {type(texts)}")
```

Comment:
dont you already do this in process_batches above so you could assert that here you deal only with lists